source/common/router/router.cc

```diff
@@ -1509,6 +1503,39 @@ bool Filter::setupRedirect(const Http::ResponseHeaderMap& headers) {
   return false;
 }
 
+bool Filter::initiateInternalRedirect(const Http::ResponseHeaderMap& headers,
+                                      const Http::HeaderEntry* location, uint64_t status_code) {
+  // Redirects are not supported for streaming requests yet.
+  if (!downstream_end_stream_) {
+    config_.stats_.passthrough_internal_redirect_stream_request_.inc();
+    return false;
+  }
+
+  if ((!internal_redirects_with_body_enabled_ || request_buffer_overflowed_) &&
+      callbacks_->decodingBuffer()) {
+    config_.stats_.passthrough_internal_redirect_body_handling_.inc();
+    return false;
+  }
+
+  if (location == nullptr) {
+    config_.stats_.passthrough_internal_redirect_bad_location_.inc();
+    return false;
+  }
+
+  // convertRequestHeadersForInternalRedirect emits stats for its failure modes, no need to
+  // increment an additional stat here.
+  if (!convertRequestHeadersForInternalRedirect(*downstream_headers_, *location, status_code)) {
```

Comment:
I think it doesn't track stats for missing path, unlikely as that is.

Reply:
I was initially unsure if it would ever get triggered but realized that it's possible in HTTP/2, huh?